nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/ConsumeSlack.java

```diff
@@ -0,0 +1,234 @@
+    .required(false)
+    .addValidator(Validator.VALID)
+    .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+    .build();
+
+  public static final PropertyDescriptor MAX_MESSAGE_QUEUE_SIZE = new PropertyDescriptor.Builder()
+    .name("Max Size of Message Queue")
+    .description("The maximum size of the internal queue used to buffer messages being transferred from the underlying channel to the processor. " +
+      "Setting this value higher allows more messages to be buffered in memory during surges of incoming messages, but increases the total " +
+      "memory used by the processor.")
+    .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+    .defaultValue("10000")
+    .required(true)
+    .build();
+
+  static final Relationship SUCCESS_RELATIONSHIP = new Relationship.Builder()
+    .name("success")
+    .description("Incoming messages that matches the given types")
+    .build();
+
+  private static final String JSON_OBJECT_TYPE_KEY = "type";
+
+  private static final List<PropertyDescriptor> DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+    SLACK_CONNECTION_SERVICE, MESSAGE_TYPES, MAX_MESSAGE_QUEUE_SIZE));
+
+  private static final Set<Relationship> RELATIONSHIPS = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+    SUCCESS_RELATIONSHIP)));
+
+  @Override
+  public Set<Relationship> getRelationships() {
+    return RELATIONSHIPS;
+  }
+
+  @Override
+  public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+    return DESCRIPTORS;
+  }
+
+  @OnScheduled
+  public void onScheduled(final ProcessContext context) {
+    String messageTypes = context.getProperty(MESSAGE_TYPES).evaluateAttributeExpressions().getValue();
+    messageTypes = messageTypes == null ? "" : messageTypes;
+    matchAny = messageTypes.isEmpty();
+    matchingTypes = Arrays.asList(messageTypes.split(","));
+    messageQueue = new LinkedBlockingQueue<>(context.getProperty(MAX_MESSAGE_QUEUE_SIZE).asInteger());
+  }
+
+  @OnUnscheduled
+  @OnShutdown
+  public void onUnscheduled() {
+    if (isProcessorRegisteredToService()) {
+      deregister();
+    }
+  }
+
+  private void deregister() {
+    slackConnectionService.deregisterProcessor(this);
+  }
+
+  @Override
+  public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+    if (!isProcessorRegisteredToService()) {
+      registerProcessorToService(context);
+    }
+
+    SlackMessage slackMessage = messageQueue.poll();
+
+    if (slackMessage != null) {
+      FlowFile flowFile = session.create();
+      session.write(flowFile, outputStream -> outputStream.write(slackMessage.getJson().getBytes()));
+      session.getProvenanceReporter().receive(flowFile, "slack");
+      session.putAttribute(flowFile, "mime.type", "application/json");
+      session.putAttribute(flowFile, "slack.type", String.valueOf(slackMessage.getType()));
+      session.transfer(flowFile, SUCCESS_RELATIONSHIP);
```

Comment:
Please add `@WritesAttributes` annotation to describe these attributes are added on the documentation.